### DISCUSSION TITLE
v0.12.42 | Interactive Printers 🎉

### DISCUSSION THREAD

**Discussion by MarvinJWendt** [pterm/pterm#371] [Announcements]
# PTerm got its own docs site! 📖

## --> https://docs.pterm.sh <--

# We now feature interactive printers! 🥳🎉
<img src="https://raw.githubusercontent.com/pterm/pterm/master/_examples/interactive_multiselect/demo/animation.svg" alt="PTerm interactive multiselect menu" width="600"/>

More here: https://docs.pterm.sh/printers/interactive

### Exciting New Features 🎉
* feat: custom writer for progressbar and spinner printers by @Brookke in https://github.com/pterm/pterm/pull/337
* Moved `NewLettersFromString` and `NewLettersFromStringWithStyle` to `putils` by @MarvinJWendt in https://github.com/pterm/pterm/pull/364
* Moved `NewBulletListXxx` to `putils` by @MarvinJWendt in https://github.com/pterm/pterm/pull/365
* Moved `NewTreeFromLeveledList` to `putils` by @MarvinJWendt in https://github.com/pterm/pterm/pull/369
* Moved `NewRGBFromHex` to `putils` by @MarvinJWendt in https://github.com/pterm/pterm/pull/370
* Added interactive printers 🎉🥳 by @MarvinJWendt in https://github.com/pterm/pterm/pull/360
### Other Changes
* ci: fixed automatic docs updates by @MarvinJWendt in https://github.com/pterm/pterm/pull/347
* new example structure by @MarvinJWendt in https://github.com/pterm/pterm/pull/349
* CI System now runs faster by @MarvinJWendt in https://github.com/pterm/pterm/pull/357

**Full Changelog**: https://github.com/pterm/pterm/compare/v0.12.41...v0.12.42

<hr /><em>This discussion was created from the release <a href='https://github.com/pterm/pterm/releases/tag/v0.12.42'>v0.12.42 | Interactive Printers 🎉</a>.</em>

**Comment by sheldonhull** [comment]
How did this not get any emoji love yet?! This is AWESOME. 

I love Bubbletea and other tools like Go-prompt, but for my use, it's rare I need prompted input. In that case, leveraging Pterm instead of bringing another library in for simple selection and input is a big win! 

Thank you all for this.

**Reply by MarvinJWendt** [reply]
Thanks a lot for the compliment! We've planned more interactive printers, like input with suggestions (autocomplete), password inputs, etc. :)
